src/C++/SocketMonitor.cpp

```diff
@@ -229,7 +229,7 @@ void SocketMonitor::processReadSet( Strategy& strategy, fd_set& readSet )
 #ifdef _MSC_VER
   for ( unsigned i = 0; i < readSet.fd_count; ++i )
   {
-    int s = readSet.fd_array[ i ];
+    socket_handle s = readSet.fd_array[ i ];
     if( s == m_interrupt )
     {
       int socket = 0;
```

Comment:
This should be socket_handle in order to match your implementation of the signal method. (I assume m_interrupt is the socket that receives the data sent by m_signal)